docs/specification/draft/client/elicitation.mdx

```diff
@@ -317,16 +521,199 @@ The three response actions are:
 
 Servers should handle each state appropriately:
 
-- **Accept**: Process the submitted data
-- **Reject**: Handle explicit rejection (e.g., offer alternatives)
+- **Accept**: Process the submitted data or proceed with the interaction
+- **Decline**: Handle explicit rejection (e.g., offer alternatives)
 - **Cancel**: Handle dismissal (e.g., prompt again later)
 
+## Implementation Considerations
+
+### Statefulness
+
+Most practical uses of elicitation require that the server maintain state about users:
+
+- Whether required information has been collected (e.g., the user's display name via form elicitation)
+- Status of resource access (e.g., API keys or a payment flow via out-of-band elicitation)
+
+Servers implementing elicitation **MUST** securely associate this state with individual users following the guidelines in the [security best practices](../basic/security_best_practices) document. Specifically:
+
+- State **MUST NOT** be associated with session IDs alone
+- User identification **MUST** be derived from authenticated tokens
```

Comment:
As mentioned below, stdio servers don't have authentication tokens of the user. They could require _an_ authentication who knows if that's actually the user of the MCP client.

Reply:
Thanks for pointing out stdio. There are a number of places in the spec that will need to be updated for stdio if we want to support that.